smallrye-reactive-messaging-mqtt-server/src/main/java/io/smallrye/reactive/messaging/mqtt/server/MqttServerSource.java

```diff
@@ -144,11 +144,7 @@ private static MqttServerOptions mqttServerOptions(Config config) {
     }
 
     synchronized void close() {
-        mqttServer.close()
-                .onFailure().invoke(t -> logger.warn("An exception has been caught while closing the MQTT server", t))
-                .onItem().invoke(x -> logger.debug("MQTT server closed"))
-                .onFailure().recoverWithItem((Void) null)
-                .await().indefinitely();
+        mqttServer.closeAndForget();
```

Comment:
With the @Priority, this should not be required anymore.

Reply:
Same question here, remove the close or go back to original code?

Reply:
`closeAndAwait`

Reply:
Should be the right code now